forge-gui/res/cardsfolder/o/once_upon_a_time.txt

```diff
@@ -1,7 +1,7 @@
 Name:Once Upon a Time
 ManaCost:1 G
 Types:Instant
-S:Mode$ Continuous | CharacteristicDefining$ True | AddKeyword$ Alternative Cost:0 | CheckSVar$ X | SVarCompare$ EQ0 | Description$ If this spell is the first spell you've cast this game, you may cast it without paying its mana cost.
+S:Mode$ AlternativeCost | ValidSA$ Spell | ValidCard$ Card.Self | ValidPlayer$ You | Cost$ 0 | EffectZone$ All | CheckSVar$ X | SVarCompare$ EQ0 | Description$ If this spell is the first spell you've cast this game, you may cast it without paying its mana cost.
```

Comment:
these ones affecting itself are broken now
probably because we were able to exclude some of STATIC_ABILITIES_SOURCE_ZONES
but now Library and Hand will probably be needed too? :/

another problem is the "You" might be the wrong player, e. g. in case of cast via _Sen Triplets_

Reply:
I added Host to StaticAbility List, we do something similar with other ones too. No need to add Library or Hand directly

as for "You", the problem is that the Alternative Cost are done before MayPlay ...
would that mean we always need to set the Controller of the card?

Reply:
Hmn, I checked and since the objects for the valids are correct another approach would be the pull the conditions into that
e. g. `ValidPlayer$ You.!castSpellThisGame` (doesn't exist)

but that involves more refactor, duplication and mixup on those parts?

Reply:
Player Property looks better imo